htdocs/contrat/class/contrat.class.php

```diff
@@ -774,6 +719,7 @@ function fetch_lines()
 				$line->fetch_optionals($line->id,$extralabelsline);
 
 				$this->lines[]			= $line;
+				$this->lines_id_index_mapper[$line->id] = key($this->lines);
```

Comment:
Because the ```$line->id``` is a low level database ID (eg. 813) and ```$this->lines``` is an ordinal array (0…n). They don't map directly to one another.
I tried indexing ``$this->lines``` with the low level ID (which would allow your suggested code) but this created a regression on other parts of the code that expected the array index to be sequential.